### PR TITLE
Make LCIO an optional dependency

### PR DESCRIPTION
**Pull request by zaborowska**
BEGINRELEASENOTES
- Make LCIO an optional dependency. If LCIO is not found, some detectors (trackers) will not be built.

ENDRELEASENOTES

The list of the detectors in CMakeLists that depend on LCIO is hard-coded and needs to be modified for each new LCIO-dependent detector.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ find_package ( ROOT REQUIRED COMPONENTS Geom GenVector)
 message ( STATUS "ROOT_VERSION: ${ROOT_VERSION}" )
 
 find_package( Geant4 REQUIRED ) 
-find_package( LCIO REQUIRED)
+find_package( LCIO QUIET)
```
Comment (andresailer) [inline CMakeLists.txt:54]:
```suggestion
OPTION(K4GEO_USE_LCIO "Enable or disable the use of LCIO, which is needed for some detector constructors and plugins" ON)
if(K4GEO_USE_LCIO)
  find_package(LCIO REQUIRED)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -76,6 +76,23 @@ file(GLOB sources
   ./detector/PID/ARC_geo_o1_v01.cpp
   )
 
+if(NOT LCIO_FOUND)
```
Comment (andresailer) [inline CMakeLists.txt:79]:
```suggestion
else() # K4GEO_USE_LCIO is OFF
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -76,6 +76,23 @@ file(GLOB sources
...
+  file(GLOB lcio_sources
```
Comment (andresailer) [inline CMakeLists.txt:80]:
```suggestion
  set(lcio_sources
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -76,6 +76,23 @@ file(GLOB sources
...
+    ./detector/tracker/TrackerBarrel_o1_v04_geo.cpp
+    ./detector/tracker/TrackerBarrel_o1_v05_geo.cpp
+    ./detector/tracker/TrackerEndcap_o1_v05_geo.cpp
+    ./detector/tracker/TrackerEndcap_o2_v06_geo.cpp
+    ./detector/tracker/VertexBarrel_detailed_o1_v01_geo.cpp
+    ./detector/tracker/VertexEndcap_o1_v05_geo.cpp
+    ./detector/tracker/ZPlanarTracker_geo.cpp
+    )
+  list(REMOVE_ITEM sources ${lcio_sources})
+  message( STATUS "LCIO has NOT been found, some detectors that depend on LCIO will not be built: ${lcio_sources}")
```
Comment (andresailer) [inline CMakeLists.txt:93]:
```suggestion
  message( STATUS "Use of LCIO is DISABLED, some detectors that depend on LCIO will not be built: ${lcio_sources}")
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -94,8 +111,13 @@ add_library(lcgeo ALIAS k4geo)
 target_include_directories(${PackageName}   PRIVATE ${PROJECT_SOURCE_DIR}/detector/include )
 target_include_directories(${PackageName}G4 PRIVATE ${PROJECT_SOURCE_DIR}/detector/include )
 
-target_link_libraries(${PackageName}   DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers ROOT::Core LCIO::lcio detectorSegmentations)
-target_link_libraries(${PackageName}G4 DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers DD4hep::DDG4 ROOT::Core ${Geant4_LIBRARIES} LCIO::lcio)
+target_link_libraries(${PackageName}   DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers ROOT::Core detectorSegmentations)
+target_link_libraries(${PackageName}G4 DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers DD4hep::DDG4 ROOT::Core ${Geant4_LIBRARIES})
+
+if(LCIO_FOUND)
```
Comment (andresailer) [inline CMakeLists.txt:117]:
```suggestion
if(K4GEO_USE_LCIO)
```

**Review by andresailer — COMMENTED** [review]
Very nice. But is is better to be explicit about the use of LCIO or not to avoid confusion when it is present or not.

**Comment by jmcarcell** [comment]
Some tests in CLICPerformance seem to be failing (from the downstream-build workflow), but they aren't related and I guess they are failing anyways:
```
lcio::IOException: [LCReader::open()] File(s) not found:  testCLIC_o3_v14.slcio  
```
so I'm going to merge this

**Pull request merged**